plasma_framework/contracts/src/exits/payment/IPaymentSpendingCondition.sol

```diff
@@ -0,0 +1,46 @@
+pragma solidity ^0.5.0;
+
+/**
+@dev only one of the two function needs to be really implemented.
+1. output type 0 --> verifyViaOwner
+2. output type not 0 --> verifyViaOutputGuard
+
+Could just left the function returns false;
+ */
+interface IPaymentSpendingCondition {
+    /**
+     * @notice The function that checks output spending condition via output guard.
+     * @param _outputGuardData The output guard data that contains info of output.
+     * @param _utxoPos (optional) serves as the identifier of output. One of utxoPos and outputId must be with value.
+     * @param _outputId (optional) serves as the identifier of output. One of utxoPos and outputId must be with value.
+     * @param _consumeTx The transaction that consumes the output.
+     * @param _inputIndex The input index of the consume transaction that points to the output.
+     * @param _witness Witness data that would be able to prove that output is able to be consumed.
+     */
+    function verifyViaOutputGuard(
+        bytes calldata _outputGuardData,
+        uint256 _utxoPos,
+        bytes32 _outputId,
+        bytes calldata _consumeTx,
+        uint8 _inputIndex,
+        bytes calldata _witness
+    ) external returns (bool);
+
+    /**
+     * @notice The function that checks output spending condition via authenticate owner.
+     * @param _authenticateOwner The authenticate user address.
+     * @param _utxoPos (optional) serves as the identifier of output. One of utxoPos and outputId must be with value.
+     * @param _outputId (optional) serves as the identifier of output. One of utxoPos and outputId must be with value.
+     * @param _consumeTx The transaction that consumes the output.
+     * @param _inputIndex The input index of the consume transaction that points to the output.
+     * @param _witness Witness data that would be able to prove that output is able to be consumed.
+     */
+    function verifyViaOwner(
+        address payable _authenticateOwner,
```

Comment:
Can we have just one `verify()` method and if necessary pass the address as bytes?
Then the SpendingCondition (or its GuardParser) will know to convert it to an address.